mwengine/src/main/cpp/audioengine.cpp

```diff
@@ -297,8 +310,8 @@ namespace MWEngine {
         int64_t expectedRenderDuration = static_cast<int64_t>(( amountOfSamplesTime * MAX_CPU_PER_RENDER_TIME ) - totalExpectedDelta );
 
 #endif
-        // erase previous buffer contents
-        inBuffer->silenceBuffers();
+        inBuffer->silenceBuffers();          // erase previous buffer contents
+        inBuffer->resize( amountOfSamples ); // keep output buffer size in sync with driver requested sample size
```

Comment:
AAudio can request render cycles for a sample amount that is lower than the initial buffer size. In order to prevent inconsistencies when rendering audio beyond the available sample amount, the output buffers have dynamic sizes which are matched to the sample amount for the render iteration.